ThingiBrowser/ThingiBrowserExtension.py

```diff
@@ -43,6 +43,8 @@ def showMainWindow(self) -> None:
         """
         if not self._main_dialog:
             self._main_dialog = self._createComponent("Thingiverse.qml")
+            if self._main_dialog:
```

Comment:
The if self._main_dialog check already exists 2 lines further. The next line (with the .connect) can simply be put after that so not another nested if is needed.

Reply:
The linter comllained because _createComponent can return None as well which is why I added the extra check.

Reply:
if you put it AFTER the next line (that already checks for it not to be None) it should be fine right?

Reply:
I am unfamiliar with pyqt signals but it would add a signal connection every time the window is reopened instead of just when the window is created? That was my line of thinking anyway.

Reply:
You're calling the `closing` signal, so it'll call resetActiveDriver every time the window is closed.

Reply:
Yes, but is it necessary to set up that listener everytime someone closes the window and then reopens it? I will defer to your decision though.

Reply:
it just occurred to me that this all only gets called on menu creation which happens once. Was getting myself confused :S